samza-api/src/main/java/org/apache/samza/system/IncomingMessageEnvelope.java

```diff
@@ -92,21 +92,55 @@ public IncomingMessageEnvelope(SystemStreamPartition systemStreamPartition, Stri
   }
 
   /**
-   * Getter for event time
+   * Getter for event time (Nanos)
    * @return this.eventTime
    */
-  public long getEventTime() {
+  public long getEventTimeNanos() {
     return eventTime;
   }
 
   /**
-   * Getter for arrival time
+   * Getter for arrival time (Nanos)
    * @return this.arrivalTime
    */
-  public long getArrivalTime() {
+  public long getArrivalTimeNanos() {
     return arrivalTime;
   }
 
+  /**
+   * For backward compatibility, Since arrivalTime was in Millis and getArrivalTime was returning
+   * arrivalTime (in Millis)
+   * @return arrivalTime in Millis
+   */
+  public long getArrivalTime() {
+    return getArrivalTimeMillis();
+  }
+
+  /**
+   * For backward compatibility, Since eventTime was in Millis and getEventTime was returning
+   * evenTime (in Millis)
+   * @return
+   */
+  public long getEventTime() {
+    return getEventTimeMillis();
+  }
+
+  /**
+   * Getter for event time in Millis
+   * @return this.eventTime / 10^6
+   */
+  public long getEventTimeMillis() {
+    return eventTime / MILLIS_IN_NANO;
```

Comment:
Please use a time util for this and below: e.g. `java.util.concurrent.TimeUnit.NANOS.convert(millis, TimeUnit.MILLIS)`

Reply:
this is no longer applicable as per our discussion